src/intrinsics/fb-www/react-native-mocks.js

```diff
@@ -0,0 +1,1675 @@
+        isTesting: false,
+        reactNativeVersion: __abstract({
+          major: 0,
+          minor: 0,
+          patch: 0,
+          prerelease: null,
+        }),
+        Version: __abstract("number"),
+        forceTouchAvailable: undefined,
+        uiMode: __abstract(),
+      }),
+      RelayAPIConfig: __abstract({
+        graphBatchURI: __abstract(),
+      }),
+      SourceCode: __abstract({
+        scriptURL: __abstract("string"),
+      }),
+    }, 'require("' + reactNameRequireName + '").NativeModules');
+
+    const {UIManager} = NativeModules;
+
+    const ReactNativeViewAttributes = {};
+    const viewConfigCallbacks = new Map();
+
+    const TextAncestor = React.createContext(false);
+
+    const ReactNativeStyleAttributes = {};
+
+    const dummySize = {width: undefined, height: undefined};
+
+    const sizesDiffer = function(one, two) {
+      one = one || dummySize;
+      two = two || dummySize;
+      return one !== two && (one.width !== two.width || one.height !== two.height);
+    };
+
+    ReactNativeStyleAttributes.transform = {process: processTransform};
+    ReactNativeStyleAttributes.shadowOffset = {diff: sizesDiffer};
+
+    const colorAttributes = {process: processColor};
+    ReactNativeStyleAttributes.backgroundColor = colorAttributes;
+    ReactNativeStyleAttributes.borderBottomColor = colorAttributes;
+    ReactNativeStyleAttributes.borderColor = colorAttributes;
+    ReactNativeStyleAttributes.borderLeftColor = colorAttributes;
+    ReactNativeStyleAttributes.borderRightColor = colorAttributes;
+    ReactNativeStyleAttributes.borderTopColor = colorAttributes;
+    ReactNativeStyleAttributes.borderStartColor = colorAttributes;
+    ReactNativeStyleAttributes.borderEndColor = colorAttributes;
+    ReactNativeStyleAttributes.color = colorAttributes;
+    ReactNativeStyleAttributes.shadowColor = colorAttributes;
+    ReactNativeStyleAttributes.textDecorationColor = colorAttributes;
+    ReactNativeStyleAttributes.tintColor = colorAttributes;
+    ReactNativeStyleAttributes.textShadowColor = colorAttributes;
+    ReactNativeStyleAttributes.overlayColor = colorAttributes;
+
+    ReactNativeViewAttributes.UIView = {
+      pointerEvents: true,
+      accessible: true,
+      accessibilityActions: true,
+      accessibilityLabel: true,
+      accessibilityComponentType: true,
+      accessibilityLiveRegion: true,
+      accessibilityTraits: true,
+      importantForAccessibility: true,
+      nativeID: true,
+      testID: true,
+      renderToHardwareTextureAndroid: true,
+      shouldRasterizeIOS: true,
+      onLayout: true,
+      onAccessibilityAction: true,
+      onAccessibilityTap: true,
+      onMagicTap: true,
+      collapsable: true,
+      needsOffscreenAlphaCompositing: true,
+      style: ReactNativeStyleAttributes,
+    };
+
+    ReactNativeViewAttributes.RCTView = Object.assign({},
+      ReactNativeViewAttributes.UIView,
+      { removeClippedSubviews: true }
+    );
+
```

Comment:
What is the source of truth for all of this? I'm sure it'll break in future RN versions.

Reply:
Like I said in other comment, it sucks, but I asked the RN folks and they pointed me to the Jest mock folder on the RN repo and said to fork and use that.